android/pytorch_android/src/main/cpp/pytorch_jni_jit.cpp

```diff
@@ -130,12 +132,40 @@ class PytorchJni : public facebook::jni::HybridClass<PytorchJni> {
     ((void)once);
   }
 
-  PytorchJni(facebook::jni::alias_ref<jstring> modelPath, jint device) {
+  PytorchJni(
+      facebook::jni::alias_ref<jstring> modelPath,
+      facebook::jni::alias_ref<
+          facebook::jni::JMap<facebook::jni::JString, facebook::jni::JString>>
+          extraFiles,
+      jint device) {
     preModuleLoadSetup();
     JITCallGuard guard;
-    module_ = torch::jit::load(std::move(modelPath->toStdString()));
-    module_.eval();
+    std::unordered_map<std::string, std::string> extra_files;
+    const auto has_extra = extraFiles && extraFiles->size() > 0;
+    if (has_extra) {
+      for (const auto& e : *extraFiles) {
+        extra_files[e.first->toStdString()] = "";
```

Comment:
What happens if the extra file doesn't exist?
Do we want to differentiate "null" state and empty state?